blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientBuilder.scala

```diff
@@ -276,14 +276,33 @@ sealed abstract class BlazeClientBuilder[F[_]] private (
 
 object BlazeClientBuilder {
 
+  sealed trait SSLContextOption
+  object SSLContextOption {
+    case object NoSSL extends SSLContextOption
+    case object TryDefaultSSLContextOrNone extends SSLContextOption
+    final case class Provided(SSLContext: SSLContext) extends SSLContextOption
+
+    def toMaybeSSLContext(sco: SSLContextOption): Option[SSLContext] =
+      sco match {
+        case SSLContextOption.NoSSL => None
+        case SSLContextOption.TryDefaultSSLContextOrNone => tryDefaultSslContext
+        case SSLContextOption.Provided(context) => Some(context)
+      }
+  }
+
+  import SSLContextOption.{TryDefaultSSLContextOrNone, toMaybeSSLContext}
+
   /** Creates a BlazeClientBuilder
     *
     * @param executionContext the ExecutionContext for blaze's internal Futures
-    * @param sslContext Some `SSLContext.getDefault()`, or `None` on systems where the default is unavailable
+    * @param sslContextOption indicates how to resolve SSLContext. The sum type offers three options:
+    *                         NoSSL                      = do not use SSL/HTTPS
+    *                         TryDefaultSSLContextOrNone = `SSLContext.getDefault()`, or `None` on systems where the default is unavailable
+    *                         Provided                   = use the explicitly passed SSLContext
     */
   def apply[F[_]: ConcurrentEffect](
       executionContext: ExecutionContext,
-      sslContext: Option[SSLContext] = tryDefaultSslContext): BlazeClientBuilder[F] =
+      sslContextOption: SSLContextOption = TryDefaultSSLContextOrNone): BlazeClientBuilder[F] =
```

Comment:
@kevinmeredith I think this is a good change, but it is a breaking change. Is this just getting merged to v1.0, or how does that work?

Reply:
> @kevinmeredith I think this is a good change, but it is a breaking change.

Good point, Keith. Per https://github.com/http4s/http4s/issues/2431#issuecomment-627715221, my expectation is that it'll become part of `1.0`, as you noted.